---
title: "*Saccharomyces cerevisiae* gene expression during fermentation of Pinot noir wines at industrially relevant scale"
always_allow_html: yes
graphics: yes
figsintext: yes
geometry: "left=1in,right=1in,top=1in,bottom=1in"
header-includes:
- \usepackage{verbatim}
- \newcommand{\comm}[1]{}
- \usepackage{float} \floatplacement{figure}{H}
- \newcommand{\beginsupplement}{\setcounter{table}{0}  \renewcommand{\thetable}{S\arabic{table}}
  \setcounter{figure}{0} \renewcommand{\thefigure}{S\arabic{figure}}}
- \usepackage[font={small,it}, labelfont={bf}]{caption}
- \usepackage{setspace}\singlespacing
- \usepackage{array}
keep_tex: yes
output:
  bookdown::pdf_document2:
    fig_caption: yes
    toc: false
---

```{r setup, include=FALSE, eval = T}
knitr::opts_chunk$set(echo = F, eval = T, message = F, warning = F, cache = T, 
                      include = T, fig.keep = 'high', fig.path = "Rmd_fig_out/",
                      dpi=400)
```

```{r libs, eval = T}
library(dplyr)
library(purrr)
library(tidyr)
library(readr)
library(tibble)
library(limma)
library(edgeR)
library(ggplot2)
library(ggthemes)
library(ggpubr)
library(clusterProfiler)
library(ggrepel)
library(kableExtra)
library(ComplexUpset)
library(cowplot)
library(ggplotify)
library(knitr)
library(stringr)
library(pheatmap)

set.seed(2)
```

```{r FUNCTIONS}
orf_to_common <- function(keys){
  common <- AnnotationDbi::select(org.Sc.sgd.db::org.Sc.sgd.db,
                                  keys = keys,
                                  columns=c("COMMON"),
                                  keytype="ORF")
  return(common)
}

orf_to_description <- function(keys){
  description <- AnnotationDbi::select(org.Sc.sgd.db::org.Sc.sgd.db,
                                  keys = keys,
                                  columns=c("DESCRIPTION"),
                                  keytype="ORF")
  return(description)
}
common_to_orf <- function(keys){
  orf <- AnnotationDbi::select(org.Sc.sgd.db::org.Sc.sgd.db,
                keys = keys,
                columns=c("ORF"),
                keytype="COMMON")
  orf <- orf$ORF
  #orf <- gsub("^Y", "sc_Y", orf)
  return(orf)
}

lm_eqn <- function(df, y, x){
    formula = as.formula(sprintf('%s ~ %s', y, x))
    m <- lm(formula, data=df);
    # formating the values into a summary string to print out
    # ~ give some space, but equal size and comma need to be quoted
    eq <- substitute(italic(target) == a + b %.% italic(input)*","~~italic(r)^2~"="~r2*","~~p~"="~italic(pvalue), 
         list(target = y,
              input = x,
              a = format(as.vector(coef(m)[1]), digits = 2), 
              b = format(as.vector(coef(m)[2]), digits = 2), 
             r2 = format(summary(m)$r.squared, digits = 3),
             # getting the pvalue is painful
             pvalue = format(summary(m)$coefficients[2,'Pr(>|t|)'], digits=1)
            )
          )
    as.character(as.expression(eq));                 
}

brix_contrast <- function(fit, coeff){
  # Test diffusion component coefficient
  con <- contrasts.fit(fit, coefficients = coeff)   
  b <- eBayes(con)
  top <- topTable(b, sort.by = "logFC", number = Inf)
  top <- top[top$adj.P.Val < .05, ]
  
  # Translate ORF to Common
  top$ORF <- rownames(top)
  commons <- orf_to_common(top$ORF)
  # subset to first occurrence of ORF
  commons <- commons[match(unique(commons$ORF), commons$ORF), ]
  commons <- commons[ , -2]
  commons$COMMON <- ifelse(is.na(commons$COMMON), commons$ORF, commons$COMMON)
  commons <- unique(commons[ , ])

  # Join Common to diffex results
  top <- left_join(top, commons, by = c("ORF"))

  # Translate ORF to DESCRIPTION
  descriptions <- orf_to_description(top$ORF)
  # subset to first occurrence of ORF
  descriptions <- descriptions[match(unique(descriptions$ORF), descriptions$ORF), ]
  descriptions <- descriptions[ , -2]
  descriptions$DESCRIPTION <- ifelse(is.na(descriptions$DESCRIPTION), descriptions$ORF, descriptions$DESCRIPTION)
  descriptions <- unique(descriptions[ , ])
  
  # Join Common to diffex results
  top <- left_join(top, descriptions, by = c("ORF"))
  
  return(top)
}



```

# Results & Discussion

## Conditions and rates of fermentation

```{r metadata}
info <- read_csv("samples_ava.csv")

# filter to 16hr, 64hr, and 112hr samples
info <- info %>%
  filter(!hr %in% c(2, 6)) %>%
  filter(!ava_id %in% c("inoculum")) %>%
  filter(!between(hr, 19, 60)) %>%
  filter(!between(hr, 84, 105)) %>%
  filter(rootstock == "101_14")
```

```{r brix_plot1, include = F}
srh = "#FFAABB" # pink
smv = "#EE8866" # orange
as  = "#EEDD88" # light yellow
snc = "#99DDFF" # light cyan 
crn = "#AAAA00" # olive
rrv = "#BBCC33" # pear 
av  = "#77AADD" # light blue

brix17 <- read_csv("../2020-pn-tmp/site_metadata/2017_fermentation_log.csv")
brix19 <- read_csv("../2020-pn-tmp/site_metadata/2019_fermentation_log.csv") 
brix <- rbind(brix17, brix19) %>%
  filter(!is.na(hours_post_innoculation))%>%
  dplyr::select(-time) %>%
  pivot_longer(cols = A:D, names_to = "tank", values_to = "brix") %>%
  mutate(ava = gsub("[123]", "", ava_id)) %>%
  filter(!ava_id %in% c("OR1", "OR2", "AV2", "RRV2", "RRV3"))
brix_plot1 <- ggplot(brix, aes(y = brix, x = hours_post_innoculation, color = ava_id, shape = tank)) +
  geom_line(alpha = .7) +
  facet_wrap(~year, ncol = 1, nrow = 2) +
  labs(x = "time (hours)", color = "vineyard", shape = "tank") +
  theme_minimal() +
  theme(legend.position = "bottom",
        strip.text.x = element_text(size = 14, face = "bold")) +
  # theme(legend.position = "bottom",
  #       axis.text.x = element_text(angle = 90),
  #       legend.title = element_text(size=12, face = "bold"),
  #       legend.text = element_text(size = 9, colour = "black"),
  #       legend.key=element_blank(),
  #       strip.text.x = element_text(size = 12, face = "bold"),
  #       strip.background = element_blank(),
  #       panel.background = element_blank(), 
  #       panel.grid =  element_blank(),
  #       panel.border = element_rect(colour = "black", fill = NA, size = 1.2)) +
  scale_color_manual(values = c(SRH1 = srh, SMV2 = smv, SMV1 = smv, AS1 = as, 
                                AS2 = as, SNC1 = snc, SNC2 = snc, CRN1 = crn, 
                                RRV1 = rrv, AV1 = av),
                     breaks = c('AV1', 'SNC1','SNC2', 'RRV1', 
                                 'CRN1', 'AS1', 'AS2', 'SMV1', 'SMV2','SRH1')) +
  scale_shape(guide = 'none')
```

```{r brix_plot2, include = F}
# note brix_plot was exported and combined with map fig in powerpoint to make figure 1
info$hr2 <- info$hr
info$hr2 <- ifelse(info$hr2 < 20, 16, info$hr2)
info$hr2 <- ifelse(info$hr2 > 20 & info$hr2 <100, 64, info$hr2)
info$hr2 <- ifelse(info$hr2 > 105, 112, info$hr2)

brix_plot2 <- ggplot(info, aes(y = brix, x = hr2)) +
  geom_point(alpha = .25) +
  theme_minimal() +
  facet_wrap(~year, nrow = 2) +
  theme(strip.text = element_text(size=12)) +
  labs(x = "time (hours)")
```

## A consistent whole-transcriptome remodeling occurs during fermentation independent of vintage

```{r counts17, include = F}
info17 <- info %>%
  filter(year == 2017)

counts17 <- read_tsv("v2017/outputs/counts/raw_counts.tsv") %>%
  filter(grepl(pattern = "^Y", x = gene)) 
colnames(counts17) <- gsub("_readcounts.txt", "", colnames(counts17))
counts17 <- counts17 %>%
  dplyr::select(c(gene, info17$id)) %>%
  as.data.frame() %>%
  column_to_rownames("gene")

# reorder counts17 to match order of info17
info17 <- info17[order(match(info17$id, colnames(counts17))), ]
```

```{r counts19, include = F}
info19 <- info %>%
  filter(year == 2019)

counts19 <- read_tsv("v2019/outputs/counts/raw_counts.tsv") %>%
  dplyr::filter(grepl(pattern = "^Y", x = gene))
colnames(counts19) <- gsub("_readcounts.txt", "", colnames(counts19))
counts19 <- counts19 %>%
  dplyr::select(c(gene, info19$id)) %>%
  as.data.frame() %>%
  column_to_rownames("gene")
# reorder counts19 to match order of info19
info19 <- info19[order(match(info19$id, colnames(counts19))), ]
```

```{r diffex17}
d0_17 <- DGEList(counts17)                    # initialize the DGE object
d17 <- calcNormFactors(d0_17)                 # calculate normalization factors
brix_mm17 <- model.matrix(~info17$brix)       # build model matrix
brix_y17 <- voom(d17, brix_mm17, plot = F)    # voom
brix_fit17 <- lmFit(brix_y17, brix_mm17)      # perform lm fit
brix_17 <- brix_contrast(fit = brix_fit17, coeff = 2) # Test brix coefficient
# filter to equivalent of log2FC of 1 over course of fermentation
brix_17$logFC_adj <- brix_17$logFC * (max(info$brix) - min(info$brix))
brix_17<- brix_17 %>%
  filter(abs(logFC_adj) > 1)

lcpm17 <- cpm(d17, log=TRUE)
```

```{r diffex19, eval = T}
d0_19 <- DGEList(counts19)              # initialize the DGE object
d19 <- calcNormFactors(d0_19)           # calculate normalization factors
brix19 <- info %>%
  dplyr::filter(year == 2019) %>%
  dplyr::filter(!is.na(brix))           # remove inoculum samples
brix_mm19 <- model.matrix(~info19$brix)
brix_y19 <- voom(d19, brix_mm19, plot = F)
brix_fit19 <- lmFit(brix_y19, brix_mm19)
brix_19 <- brix_contrast(fit = brix_fit19, coeff = 2) # Test brix coefficient
# filter to equivalent of log2FC of 1 over course of fermentation
brix_19$logFC_adj <- brix_19$logFC * (max(info$brix) - min(info$brix))
brix_19 <- brix_19 %>%
  filter(abs(logFC_adj) > 1)

# get counts as log cpm
lcpm19 <- cpm(d19, log=TRUE)
```

```{r brix_upset, include = F}

induced_v2017   <- filter(brix_17, logFC > 0)
repressed_v2017 <- filter(brix_17, logFC < 0)
induced_v2019   <- filter(brix_19, logFC > 0) 
repressed_v2019 <- filter(brix_19, logFC < 0)

brix_list <- list("induced late 2019" = repressed_v2019$ORF,
                  "induced early 2019"  = induced_v2019$ORF,
                  "induced late 2017" = repressed_v2017$ORF,
                  "induced early 2017"   = induced_v2017$ORF)
brix_list_upset <- UpSetR::fromList(brix_list)
intersects <- c("induced late 2019", "induced early 2019", "induced late 2017", "induced early 2017")
brix_list_upset[intersects] = brix_list_upset[intersects] == 1
upset(brix_list_upset, intersects, width_ratio = 0.1,
                    sort_intersections_by='cardinality', height_ratio = 1, 
                    dot_size = 2, set_sizes=FALSE, name=NULL, sort_sets = F,
                    base_annotations=list('Intersection size'=intersection_size(text=list(size = 3.2)))) & theme(plot.background=element_rect(fill='transparent', color=NA))
```

```{r calc_intersect}
induced_intersect17 <- induced_v2017[induced_v2017$ORF %in% induced_v2019$ORF, ] %>%
  dplyr::select(ORF, COMMON, DESCRIPTION, logFC17 = logFC, logFC_adj17 = logFC_adj)
repressed_intersect17 <- repressed_v2017[repressed_v2017$ORF %in% repressed_v2019$ORF, ] %>%
  dplyr::select(ORF, COMMON, DESCRIPTION, logFC17 = logFC, logFC_adj17 = logFC_adj)
induced_intersect19 <- induced_v2019[induced_v2019$ORF %in% induced_v2017$ORF, ] %>%
  dplyr::select(ORF, COMMON, DESCRIPTION, logFC19 = logFC, logFC_adj19 = logFC_adj)
repressed_intersect19 <- repressed_v2019[repressed_v2019$ORF %in% repressed_v2017$ORF, ] %>%
  dplyr::select(ORF, COMMON, DESCRIPTION, logFC19 = logFC, logFC_adj19 = logFC_adj)

induced <- left_join(induced_intersect17, induced_intersect19) %>%
  mutate(logFC_avg = (logFC17 + logFC19)/2) %>%
  mutate(logFC_adj_avg = (logFC_adj17 + logFC_adj19)/2)

repressed <- left_join(repressed_intersect17, repressed_intersect19) %>%
  mutate(logFC_avg = (logFC17 + logFC19)/2) %>%
  mutate(logFC_adj_avg = (logFC_adj17 + logFC_adj19)/2)
```

```{r gene_examples, include = F}
gene = "YHR094C" # hxt1
Y <- brix_y17$E[gene, ]
info17$Y = Y
# calc log fc
logfc <- brix_17 %>% 
  filter(ORF == gene) %>% 
  dplyr::select(logFC_adj) %>% 
  mutate(logFC_adj = round(logFC_adj, digits = 2))

hxt1_y2017 <- ggplot(info17, aes(y = Y, x = brix)) +
  geom_point(color = "black", alpha = .7) +
  labs(y = "norm log counts",
       x = "",
       title = expression(paste(italic("HXT1"))),
       subtitle = "2017",
       caption =expression(paste(log[2], "FC = 4.3"))) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        plot.subtitle = element_text(hjust = 0.5),
        axis.title.x = element_blank()) +
  geom_smooth(method='lm', color = "grey", alpha = .7, se= F)  +
  ylim(c(0, 13))


Y <- brix_y19$E[gene, ]
info19$Y = Y
logfc <- brix_19 %>% 
  filter(ORF == gene) %>% 
  dplyr::select(logFC_adj) %>% 
  mutate(logFC_adj =  round(logFC_adj, digits = 2))

hxt1_y2019 <- ggplot(info19, aes(y = Y, x = brix)) +
  geom_point(color = "black", alpha = .7) +
  labs(y = "norm log counts",
       title = "",
       subtitle = "2019",
       caption =expression(paste(log[2], "FC = 3.25"))) +
  theme_minimal() +
  theme(plot.title = element_blank(),
        plot.subtitle = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', color = "grey", alpha = .7, se= F)  +
  ylim(c(0, 13))

# HXT4:
gene = "YHR092C"
Y <- brix_y17$E[gene, ]
info17$Y = Y
logfc <- brix_17 %>% 
  filter(ORF == gene) %>% 
  dplyr::select(logFC_adj) %>% 
  mutate(logFC_adj = -1* round(logFC_adj, digits = 2))

hxt4_y2017 <- ggplot(info17, aes(y = Y, x = brix)) +
  geom_point(color = "black", alpha = .7) +
  labs(y = "",
       x = "",
       title = expression(paste(italic("HXT4"))),
       subtitle = "2017",
       caption =expression(paste(log[2], "FC = 6.8"))) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        plot.subtitle = element_text(hjust = 0.5),
        axis.title.x = element_blank()) +
  geom_smooth(method='lm', color = "grey", alpha = .7, se= F)  +
  ylim(c(0, 13))


Y <- brix_y19$E[gene, ]
info19$Y = Y
logfc <- brix_19 %>% 
  filter(ORF == gene) %>% 
  dplyr::select(logFC_adj) %>% 
  mutate(logFC_adj = -1* round(logFC_adj, digits = 2))

hxt4_y2019 <- ggplot(info19, aes(y = Y, x = brix)) +
  geom_point(color = "black", alpha = .7) +
  labs(y = "",
       title = "",
       subtitle = "2019",
       caption =expression(paste(log[2], "FC = 5.3"))) +
  theme_minimal() +
  theme(plot.title = element_blank(),
        plot.subtitle = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', color = "grey", alpha = .7, se= F)  +
  ylim(c(0, 13))

# ADH1
# gene = 'YGR240C' # PFK1
gene = "YOL086C" # ADH1
Y <- brix_y17$E[gene, ]
info17$Y = Y
logfc <- brix_17 %>% 
  filter(ORF == gene) %>% 
  dplyr::select(logFC_adj) %>% 
  mutate(logFC_adj = -1* round(logFC_adj, digits = 2))

adh1_y2017 <- ggplot(info17, aes(y = Y, x = brix)) +
  geom_point(color = "black", alpha = .7) +
  labs(y = "",
       x = "",
       title = expression(paste(italic("ADH1"))),
       subtitle = "2017") +
       #caption =expression(paste(log[2], "FC = 1.1"))) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        plot.subtitle = element_text(hjust = 0.5),
        axis.title.x = element_blank()) +
  geom_smooth(method='lm', color = "grey", alpha = .7, se= F)  +
  ylim(c(0, 15))


Y <- brix_y19$E[gene, ]
info19$Y = Y
logfc <- brix_19 %>% 
  filter(ORF == gene) %>% 
  dplyr::select(logFC_adj) %>% 
  mutate(logFC_adj = -1* round(logFC_adj, digits = 2))

adh1_y2019 <- ggplot(info19, aes(y = Y, x = brix)) +
  geom_point(color = "black", alpha = .7) +
  labs(y = "",
       title = "",
       subtitle = "2019") +
       #caption =expression(paste(log[2], "FC = 5.3"))) +
  theme_minimal() +
  theme(plot.title = element_blank(),
        plot.subtitle = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', color = "grey", alpha = .7, se= F)  +
  ylim(c(0, 15))

ggarrange(hxt1_y2017, hxt4_y2017, adh1_y2017,
          hxt1_y2019, hxt4_y2019, adh1_y2019, 
          #labels = c("A", "B", "C", "D", "E", "F"), 
          nrow = 2, ncol = 3)
```

```{r write_diffex_out, eval = F}
write_tsv(induced, "supplementary_tables/avg_induced.tsv")
write_tsv(repressed, "supplementary_tables/avg_repressed.tsv")
write_tsv(induced_v2017, "supplementary_tables/induced_2017.tsv")
write_tsv(repressed_v2017, "supplementary_tables/repressed_2017.tsv")
write_tsv(induced_v2019, "supplementary_tables/induced_2019.tsv")
write_tsv(repressed_v2019, "supplementary_tables/repressed_2019.tsv")

library(xlsx)
write.xlsx(induced, "supplementary_tables/table_s1.xlsx", sheetName = "avg_induced", row.names = F)
write.xlsx(repressed, "supplementary_tables/table_s1.xlsx", sheetName = "avg_repressed", append = T, row.names = F)
write.xlsx(induced_v2017, "supplementary_tables/table_s1.xlsx", sheetName = "induced_2017", append = T, row.names = F)
write.xlsx(repressed_v2017, "supplementary_tables/table_s1.xlsx", sheetName = "repressed_2017", append = T, row.names = F)
write.xlsx(induced_v2019, "supplementary_tables/table_s1.xlsx", sheetName = "induced_2019", append = T, row.names = F)
write.xlsx(repressed_v2019, "supplementary_tables/table_s1.xlsx", sheetName = "repressed_2019", append = T, row.names = F)
```

```{r write_for_proteomap, eval = F}
write_tsv(induced %>%
            select(ORF, logFC_adj_avg),
          "supplementary_tables/proteomaps/avg_induced_for_proteomap.tsv", 
          col_names = F)

write_tsv(repressed %>%
            select(ORF, logFC_adj_avg) %>%
            mutate(logFC_adj_avg = logFC_adj_avg * -1),
          "supplementary_tables/proteomaps/avg_repressed_for_proteomap.tsv", 
          col_names = F)

write_tsv(induced_v2017  %>%
            select(ORF, logFC_adj), 
          "supplementary_tables/proteomaps/induced_2017_for_proteomaps.tsv",
          col_names = F)
write_tsv(repressed_v2017  %>%
            select(ORF, logFC_adj) %>%
            mutate(logFC_adj = logFC_adj * -1),  
          "supplementary_tables/proteomaps/repressed_2017_for_proteomaps.tsv",
          col_names = F)
write_tsv(induced_v2019 %>%
            select(ORF, logFC_adj), 
          "supplementary_tables/proteomaps/induced_2019_for_proteomaps.tsv",
          col_names = F)
write_tsv(repressed_v2019 %>%
            select(ORF, logFC_adj) %>%
            mutate(logFC_adj = logFC_adj * -1),  
          "supplementary_tables/proteomaps/repressed_2019_for_proteomaps.tsv",
          col_names = F)

```

## Nutrient limitation in early fermentation

## Mitochondria and fermentation

```{r, eval = F}
# helper code to investigate based on name
tmp <- orf_to_description(repressed$ORF)
repressed$DESCRIPTION <- tmp$DESCRIPTION
repressed %>%
  filter(str_detect(string = DESCRIPTION, pattern = "metallothio"))
```

### Thioredoxins and glutathione system activity throughout fermentation

### Stress-induced gene expression compared to other stress responses

```{r iesr, eval = F}
esr <- read_csv("reference_genes/iesr.csv")
table(induced$ORF %in% esr$ORF)
table(repressed$ORF %in% esr$ORF)
#View(repressed[repressed$ORF %in% esr$ORF, ])
#View(induced[induced$ORF %in% esr$ORF, ])

enrich_ind <- enrichGO(induced[induced$COMMON %in% esr$COMMON, ]$COMMON, OrgDb = org.Sc.sgd.db::org.Sc.sgd.db,
                             keyType = "GENENAME", ont = "BP")
plt1 <- dotplot(enrich_ind, font.size = 8) + 
  geom_point(color = "black")


enrich_rep <- enrichGO(repressed[repressed$COMMON %in% esr$COMMON, ]$COMMON, OrgDb = org.Sc.sgd.db::org.Sc.sgd.db,
                             keyType = "GENENAME", ont = "BP")
plt2 <-  dotplot(enrich_rep, font.size = 8) + 
  geom_point(color = "black")

ggarrange(plt1, plt2, common.legend = T, legend = "bottom")
```

```{r fsr, eval = F}
fsr <- read_csv("reference_genes/marks2008_fsr.csv", col_types = c("ncccnccnnnnnnn"))
table(induced$COMMON %in% fsr$Gene)
table(repressed$COMMON %in% fsr$Gene)
repressed[repressed$COMMON %in% fsr$Gene, ]
#View(fsr[fsr$Gene %in% repressed$COMMON, ])

fsr_pn <- fsr[fsr$Gene %in% repressed$COMMON, ]
fsr_pn$other <- rowSums(fsr_pn[ , 8:14], na.rm = T)
sum(fsr_pn$other == 0)/nrow(fsr_pn)

fsr_pn_distinct <- fsr_pn %>%
  filter(other == 0)

fsr_pn_distinct_go<- enrichGO(fsr_pn_distinct$COMMON, OrgDb = org.Sc.sgd.db::org.Sc.sgd.db,
                             keyType = "GENENAME", ont = "ALL")
dotplot(fsr_pn_distinct_go)

fsr_pn_distinct_kegg <- enrichKEGG(fsr_pn_distinct$ORF, organism = "sce")
dotplot(fsr_pn_distinct_kegg)
fsr_pn_distinct$description <- orf_to_description(fsr_pn_distinct$ORF)

# overlap with CER/ESR
# sum(fsr_pn$ESR, na.rm = T)
# sum(fsr_pn$CER, na.rm = T)
# sum(fsr$CER, na.rm = T)
```


### Polyol metabolism in late fermentation

```{r mannitolplt, eval = F}
gene = "YNR073C"
Y <- brix_y17$E[gene, ]
info17$Y = Y
# calc log fc
logfc <- brix_17 %>% 
  filter(ORF == gene) %>% 
  dplyr::select(logFC_adj) %>% 
  mutate(logFC_adj = -1* round(logFC_adj, digits = 2))

man2_y2017 <- ggplot(info17, aes(y = Y, x = brix)) +
  geom_point(color = "black", alpha = .7) +
  labs(y = "",
       x = "",
       title = expression(paste(italic("MAN2"))),
       subtitle = "2017",
       caption =expression(paste(log[2], "FC = 5.9"))) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        plot.subtitle = element_text(hjust = 0.5),
        axis.title.x = element_blank()) +
  geom_smooth(method='lm', color = "grey", alpha = .7, se= F)  +
  ylim(c(-3, 13))


Y <- brix_y19$E[gene, ]
info19$Y = Y
logfc <- brix_19 %>% 
  filter(ORF == gene) %>% 
  dplyr::select(logFC_adj) %>% 
  mutate(logFC_adj = -1* round(logFC_adj, digits = 2))

man2_y2019 <- ggplot(info19, aes(y = Y, x = brix)) +
  geom_point(color = "black", alpha = .7) +
  labs(y = "",
       title = "",
       subtitle = "2019",
       caption =expression(paste(log[2], "FC = 6.74"))) +
  theme_minimal() +
  theme(plot.title = element_blank(),
        plot.subtitle = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', color = "grey", alpha = .7, se= F)  +
  ylim(c(-3, 13))

# HXT13:
gene = "YEL069C"
Y <- brix_y17$E[gene, ]
info17$Y = Y
logfc <- brix_17 %>% 
  filter(ORF == gene) %>% 
  dplyr::select(logFC_adj) %>% 
  mutate(logFC_adj = -1* round(logFC_adj, digits = 2))

hxt13_y2017 <- ggplot(info17, aes(y = Y, x = brix)) +
  geom_point(color = "black", alpha = .7) +
  labs(y = "norm log counts",
       x = "",
       title = expression(paste(italic("HXT13"))),
       subtitle = "2017",
       caption =expression(paste(log[2], "FC = 6.4"))) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5),
        plot.subtitle = element_text(hjust = 0.5),
        axis.title.x = element_blank()) +
  geom_smooth(method='lm', color = "grey", alpha = .7, se= F)  +
  ylim(c(-3, 13))


Y <- brix_y19$E[gene, ]
info19$Y = Y
logfc <- brix_19 %>% 
  filter(ORF == gene) %>% 
  dplyr::select(logFC_adj) %>% 
  mutate(logFC_adj = -1* round(logFC_adj, digits = 2))

hxt13_y2019 <- ggplot(info19, aes(y = Y, x = brix)) +
  geom_point(color = "black", alpha = .7) +
  labs(y = "norm log counts",
       title = "",
       subtitle = "2019",
       caption =expression(paste(log[2], "FC = 7.26"))) +
  theme_minimal() +
  theme(plot.title = element_blank(),
        plot.subtitle = element_text(hjust = 0.5)) +
  geom_smooth(method='lm', color = "grey", alpha = .7, se= F)  +
  ylim(c(-3, 13))

ggarrange(hxt13_y2017, man2_y2017,  hxt13_y2019, man2_y2019,labels = c("A", "B", "C", "D"), 
          nrow = 2, ncol = 2)
```

### Vintage-specific differences

```{r unique, eval = F}
all <- rbind(induced, repressed)

brix_17_unique <- brix_17 %>%
  filter(!ORF %in% all$ORF)

brix_17_unique_down <- brix_17 %>%
  filter(!ORF %in% all$ORF) %>%
  filter(logFC < 0)

brix_17_unique_up <- brix_17 %>%
  filter(!ORF %in% all$ORF) %>%
  filter(logFC > 0)

brix_19_unique <- brix_19 %>%
  filter(!ORF %in% all$ORF) 

brix_19_unique_down <- brix_19 %>%
  filter(!ORF %in% all$ORF) %>%
  filter(logFC < 0)

brix_19_unique_up <- brix_19 %>%
  filter(!ORF %in% all$ORF) %>%
  filter(logFC > 0)

dotplot(enrichKEGG(gene = brix_17_unique$ORF, organism = 'sce'))

tmp <- enrichKEGG(gene = brix_19_unique_down$ORF, organism = 'sce')
dotplot(tmp)

tmp <- enrichKEGG(gene = brix_17_unique_up$ORF, organism = 'sce')
dotplot(tmp)


tmp <- enrichGO(brix_17_unique_up$COMMON, OrgDb = org.Sc.sgd.db::org.Sc.sgd.db,
         keyType = "GENENAME", ont = "ALL")
dotplot(tmp)
tmp <- enrichGO(brix_17_unique_down$COMMON, OrgDb = org.Sc.sgd.db::org.Sc.sgd.db,
         keyType = "GENENAME", ont = "ALL")
dotplot(tmp)
tmp <- enrichGO(brix_19_unique_up$COMMON, OrgDb = org.Sc.sgd.db::org.Sc.sgd.db,
         keyType = "GENENAME", ont = "ALL")
dotplot(tmp)
tmp <- enrichGO(brix_19_unique_down$COMMON, OrgDb = org.Sc.sgd.db::org.Sc.sgd.db,
         keyType = "GENENAME", ont = "ALL")
dotplot(tmp)
```

```{r vintage_unknown, eval = F}
unk17 <- brix_17_unique %>%
  filter(abs(logFC_adj) > 2) %>%
  filter(grepl("^Y", COMMON))

unk19 <- brix_19_unique %>%
  filter(abs(logFC_adj) > 2) %>%
  filter(grepl("^Y", COMMON))
```

\newpage
# Supplementary material {-}

\beginsupplement


```{r enrichmentindgoplt, eval = F, fig.height = 8}
induced_enrich_go <- enrichGO(induced$COMMON, OrgDb = org.Sc.sgd.db::org.Sc.sgd.db,
                             keyType = "GENENAME", ont = "ALL", pAdjustMethod = "bonferroni")
induced_enrich_go_res <- induced_enrich_go@result
plot_rows <- induced_enrich_go_res %>%
  filter(p.adjust < .05) %>%
  nrow()
dotplot(induced_enrich_go, showCategory = plot_rows, font.size = 4)
```

```{r enrichmentrepgoplt, eval = F,fig.height = 6}
repressed_enrich_go <- enrichGO(repressed$COMMON, OrgDb = org.Sc.sgd.db::org.Sc.sgd.db,
                             keyType = "GENENAME", ont = "ALL", pAdjustMethod = "bonferroni")
repressed_enrich_go_res <- repressed_enrich_go@result
plot_rows <- repressed_enrich_go_res %>%
  filter(p.adjust < .05) %>%
  nrow()
dotplot(repressed_enrich_go, showCategory = plot_rows, font.size = 7)
```

```{r enrichmentindkeggplt, eval = F}
## KEGG
induced_enrich_kegg <- enrichKEGG(induced$ORF, organism = "sce", pAdjustMethod = "bonferroni")
induced_enrich_kegg_res <- induced_enrich_kegg@result
plot_rows <- induced_enrich_kegg_res %>%
  filter(p.adjust < .05) %>%
  nrow()
dotplot(induced_enrich_kegg, showCategory = plot_rows, font.size = 12)
```

```{r enrichmentrepkeggplt, eval = F}
repressed_enrich_kegg <- enrichKEGG(repressed$ORF, organism = "sce", pAdjustMethod = "bonferroni")

repressed_enrich_kegg_res <- repressed_enrich_kegg@result
plot_rows <- repressed_enrich_kegg_res %>%
  filter(p.adjust < .05) %>%
  nrow()
dotplot(repressed_enrich_kegg, showCategory = plot_rows, font.size = 12)
```